src/project_types.rs

```diff
@@ -80,4 +108,135 @@ impl Project {
+            Ok(new_project)
+        }
+    }
+
+    fn duplicate_sequence(&self, name: &str) -> Error {
+        Error::DuplicateSequence(name.to_string())
+    }
+
 }
+
+impl Sequence {
+    /// Creates a new Sequence, allowing default values
+    /// Also initializes section files
+    pub fn new(name: &str,
+        seq_directory_name: &str,
+        music_file_name: &str,
+        music_duration_sec: u32,
+        frame_duration_ms: Option<u32>,
+        num_sections: Option<u32>
+    ) -> Result<Sequence, Error> {
+        // Defaults
+        let frame_dur_ms = match frame_duration_ms {
+            Some(duration) => duration,
+            None => 50,
+        };
+        let num_sects = match num_sections {
+            Some(num) => num,
+            None => 1,
+        };
+
+        // Create sequence
+        let sequence = Sequence {
+            name: name.to_string(),
+            directory_name: seq_directory_name.to_string(),
+            music_file_name: music_file_name.to_string(),
+            music_duration_sec: music_duration_sec,
+            frame_duration_ms: frame_dur_ms,
+            num_sections: num_sects,
+        };
+
+        // Create section files
+        if num_sects == 1 {
+            let num_frames_f32: f32 = (music_duration_sec as f32 * 1000_f32) / frame_dur_ms as f32;
+            let num_frames = num_frames_f32.ceil() as u32;
+            let num_channels = 1; // TODO: change when add layout
+            let sequence_section = SequenceSection {
+                seq_name: name.to_string(),
+                index: 1,
+                num_frames: num_frames,
+                editor: None,
+                data: vec![vec![0; num_frames as usize]; num_channels],
+            };
+            try!(sequence_section.write_to_file(&seq_directory_name));
+        } else {
+            try!(sequence.resection(num_sects));
+        }
+
+        Ok(sequence)
+    }
+
+    /// Resection a sequence
+    pub fn resection(&self, num_sections: u32) -> Result<(), Error> {
+        Err(Error::TodoErr)
+    }
+
+
+}
+
+impl SequenceSection {
+    /// Write the sequence section to a file
+    pub fn write_to_file(&self, seq_directory_name: &str) -> Result<(), Error> {
+        let pretty_json = json::as_pretty_json(&self);
+        let section_path = &self.get_path(&seq_directory_name);
+
+        File::create(&section_path)
+            .and_then(|mut section_file| write!(section_file, "{}\n", pretty_json))
+            .map_err(Error::Io)
+    }
+
+    /// Get the path to this specific section, starting with the sequence directory
+    /// E.g. sequence/sequence_section1.json
+    /// Assumes the current directory is the project directory
+    fn get_path(&self, directory_name: &str) -> PathBuf {
+
+        let mut file_name = String::new();
+        file_name.push_str(&self.seq_name);
+        file_name.push_str("_section");
+        file_name.push_str(&self.index.to_string());
+        let file_name = file_name;
```

Comment:
I think this is unnecessary. I get you're declaring that filename is no longer mutable, but it just seems pointless. Also, use filename instead of file_name.